src/components/Payroll/PayrollLanding/PayrollLanding.test.tsx

```diff
@@ -0,0 +1,190 @@
+    })
+
+    it('switches back to run payroll component when tab is clicked', async () => {
+      const user = userEvent.setup()
+      renderWithProviders(<PayrollLanding {...defaultProps} />)
+
+      // Wait for initial render
+      await waitFor(() => {
+        expect(screen.getByTestId('run-payroll')).toBeInTheDocument()
+      })
+
+      // Switch to payroll history tab
+      const payrollHistoryTab = screen.getByRole('tab', { name: 'Payroll history' })
+      await user.click(payrollHistoryTab)
+
+      await waitFor(() => {
+        expect(screen.getByTestId('payroll-history')).toBeInTheDocument()
+      })
+
+      // Switch back to run payroll tab
+      const runPayrollTab = screen.getByRole('tab', { name: 'Run payroll' })
+      await user.click(runPayrollTab)
+
+      // Check that run payroll component is visible again
+      await waitFor(() => {
+        expect(screen.getByTestId('run-payroll')).toBeInTheDocument()
+      })
+
+      expect(screen.getByText('Run Payroll')).toBeInTheDocument()
+    })
+  })
+
+  describe('accessibility', () => {
+    it('renders with proper accessibility structure', async () => {
+      renderWithProviders(<PayrollLanding {...defaultProps} />)
+
+      await waitFor(() => {
+        expect(screen.getByTestId('run-payroll')).toBeInTheDocument()
+      })
+
+      // Verify the component renders without accessibility violations
+      // The tab component itself should handle its own accessibility
+      expect(screen.getByText('Run Payroll')).toBeInTheDocument()
+    })
+  })
+
+  describe('event handling', () => {
+    it('passes onEvent to child components', async () => {
+      renderWithProviders(<PayrollLanding {...defaultProps} />)
+
+      await waitFor(() => {
+        expect(screen.getByTestId('run-payroll')).toBeInTheDocument()
+      })
+
+      // Click the test event button in the run payroll component
+      const testButton = screen.getByText('Test Event')
+      await userEvent.click(testButton)
+
+      // Check that the event was called
+      expect(defaultProps.onEvent).toHaveBeenCalledWith('test-event', {})
+    })
+
+    it('passes companyId to child components', async () => {
+      renderWithProviders(<PayrollLanding {...defaultProps} />)
+
+      await waitFor(() => {
+        expect(screen.getByText('Company ID: test-company-123')).toBeInTheDocument()
+      })
+    })
+  })
+
+  describe('internationalization', () => {
+    it('uses correct translation keys for tab labels', async () => {
+      renderWithProviders(<PayrollLanding {...defaultProps} />)
+
+      await waitFor(() => {
+        expect(screen.getByTestId('run-payroll')).toBeInTheDocument()
+      })
+
+      // Check that translated tab labels are used
+      expect(screen.getByRole('tab', { name: 'Run payroll' })).toBeInTheDocument()
+      expect(screen.getByRole('tab', { name: 'Payroll history' })).toBeInTheDocument()
+    })
+  })
```

Comment:
Nit: this kinda gets tested implicitly already with the above since we query by these to select the tabs